bravado/client.py

```diff
@@ -457,11 +503,25 @@ def pick_a_scheme(schemes):
     return urlparse.urlunparse((scheme, netloc, path, None, None, None))
 
 
-def build_resources_from_spec(http_client, spec, api_base_path, url_base):
-    return dict(
-        (api_doc['name'],
-         Resource.from_api_doc(api_doc, http_client, api_base_path, url_base))
-        for api_doc in spec['paths'])
+def build_resources(spec, http_client):
+    """Transforms the REST resources in the json-like spec into rich :Resource:
+    objects that have associated :Operation:s.
+
+    :param spec: json-like spec in dict form
+    :param http_client: an HTTP client used to perform requests
+    :type  http_client: :class:`bravado.http_client.HttpClient`
+    :returns: dict where (key,value) = (resource name, Resource instance)
+    """
+    resources = {}
+    paths = spec['paths']
+    for path_name, path_dict in paths:
+        resources[path_name] = Resource.from_path(
+            path_name,
+            path_dict,
+            spec['x_api_url'],
+            http_client)
+
+    return resources
```

Comment:
I'm a fan of comprehensions for these kind of things, but I guess that's a personal style preference. This works too